### PR TITLE
add eastl/3.16.05 + fix min version for vc

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **add eastl/3.16.05**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This closes https://github.com/conan-io/conan-center-index/issues/552, and closes https://github.com/conan-io/conan-center-index/issues/553

### CODE REVIEW

**Comment by SpaceIm** [comment]
The recipe was broken for Visual Studio as a shared lib (and didn't hide symbols for GNU compilers), but it passed CI so far due to broken test on Visual Studio compiler version.

File: recipes/eastl/all/conanfile.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def configure(self):
 
         if (self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5") or \
                 (self.settings.compiler == "clang" and Version(self.settings.compiler.version.value) < "3.4") or \
-                (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version.value) < "2015"):
+                (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version.value) < "14"):
```
Comment (uilianries) [inline recipes/eastl/all/conanfile.py]: you can use `Version(self.settings.compiler.version)`, but it's okay.

Comment (SpaceIm) [reply recipes/eastl/all/conanfile.py]: fixed

File: recipes/eastl/all/conanfile.py [hunk 2]
```diff
@@ -76,5 +90,7 @@ def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "EASTL"
         self.cpp_info.names["cmake_find_package_multi"] = "EASTL"
         self.cpp_info.libs = ["EASTL"]
-        if self.settings.os in ("Android", "Linux", "Macos", "watchOS", "tvOS"):
+        if self.settings.os == "Linux":
```
Comment (uilianries) [inline recipes/eastl/all/conanfile.py]: are you sure about Android? Asking because I have no idea about

Comment (SpaceIm) [reply recipes/eastl/all/conanfile.py]: No, but AFAIK linking pthread on Android is useless.

**Comment by SpaceIm** [comment]
I think I have fixed issue https://github.com/conan-io/conan-center-index/pull/554 tried to adress.

**Pull request merged**